mmdet3d/models/necks/view_transformer.py

```diff
@@ -0,0 +1,347 @@
+
+    Args:
+        x (torch.tensor): Point-wise features in shape (N_Points, C).
+        coor (torch.tensor): of shape (N_Points, D). The coordinate of points
+            in the feature coordinate system.
+        ranks (torch.tensor): of shape (N_Points). The rank of voxel that a
+            point is belong to.
+
+    Returns:
+        torch.tensor: Voxel-wise features in shape (N_Voxels, C).
+        torch.tensor: of shape (N_Voxels,3). The coordinate of voxels in the
+            feature coordinate system.
+    """
+
+    @staticmethod
+    def forward(ctx, x, coor, ranks):
+        x = x.cumsum(0)
+        kept = torch.ones(x.shape[0], device=x.device, dtype=torch.bool)
+        kept[:-1] = (ranks[1:] != ranks[:-1])
+
+        x, coor = x[kept], coor[kept]
+        x = torch.cat((x[:1], x[1:] - x[:-1]))
+
+        # save kept for backward
+        ctx.save_for_backward(kept)
+
+        # no gradient for coordinates
+        ctx.mark_non_differentiable(coor)
+
+        return x, coor
+
+    @staticmethod
+    def backward(ctx, gradx, gradcoor):
+        kept, = ctx.saved_tensors
+        back = torch.cumsum(kept, 0)
+        back[kept] -= 1
+
+        val = gradx[back]
+
+        return val, None, None
+
+
+@NECKS.register_module()
+class ViewTransformerLSS(BaseModule):
+    r"""Lift-Splat-Shoot view transformer.
+
+    Please refer to the `paper <https://arxiv.org/abs/2008.05711>`_
+
+    Args:
+        grid_config (dict): Config of grid alone each axis in format of
+            (lower_bound, upper_bound, interval). axis in {x,y,z,depth}.
+        input_size (tuple(int)): Size of input images in format of (height,
+            width).
+        in_channels (int): Channels of input feature.
+        out_channels (int): Channels of transformed feature.
+        accelerate (bool): Whether the view transformation is conducted with
+            acceleration. Note: the intrinsic and extrinsic of cameras should
+            be constant when 'accelerate' is set true.
+        max_voxel_points (int): Specify the maximum point number in a single
+            voxel during the acceleration.
+    """
+
+    def __init__(self,
+                 grid_config,
+                 input_size,
+                 downsample,
+                 in_channels,
+                 out_channels,
+                 accelerate=False,
+                 max_voxel_points=300):
+        super(ViewTransformerLSS, self).__init__()
+        self.gen_grid_infos(**grid_config)
+
+        self.input_size = input_size
+        self.downsample = downsample
+
+        self.create_frustum(grid_config['depth'])
+        self.out_channels = out_channels
+        self.depth_net = nn.Conv2d(
+            in_channels, self.D + self.out_channels, kernel_size=1, padding=0)
+        self.accelerate = accelerate
+        self.max_voxel_points = max_voxel_points
+        self.initial_flag = True
+
+    def gen_grid_infos(self, x, y, z, **kwargs):
```

Comment:
would better be renamed as create_grid to be consistent with create_frustum?

Reply:
create_grid_infos? it is not for creating grids but the information of the grids instead.